osm2gtfs/tests/creators/tests_creators_abstract.py

```diff
@@ -0,0 +1,326 @@
+import unittest
+import os
+import zipfile
+import csv
+import overpy
+import transitfeed
+
+from mock import patch
+from osm2gtfs.core.configuration import Configuration
+from osm2gtfs.core.osm_connector import OsmConnector
+from osm2gtfs.core.creator_factory import CreatorFactory
+from osm2gtfs.core.cache import Cache
+
+current_dir = os.path.dirname(__file__)
+
+
+class TestCreatorsArgs():
+    def __init__(self, config_file, selector):
+        """
+        Prepare arguments for the Initialization of an Configuration object
+        (see osm2gtfs.core.configuration for more information)
+
+        """
+        # Basic preparations
+        self.config = open(config_file)
+        self.selector = selector
+
+        # Overriding the output to not interfere with user's data
+        self.output = os.path.realpath(
+            os.path.join(current_dir, "../../../data/" + self.selector + "_tests.zip"))
+
+
+class TestCreatorsAbstract(unittest.TestCase):
+    def setUp(self):
+        """
+        This method sets the basic variables for testing of providers.
+        It needs to be implemented for each provider in the heriting class.
+        """
+
+        # Define selector for this tests
+        self.selector = "YOUR SELECTOR"
```

Comment:
Maybe this would be better with an abstract method?

Reply:
This is where my knowledge on Python ends currently. [I gave it a try here](https://github.com/xamanu/osm2gtfs/blob/creator-tests-abstract-playground/osm2gtfs/tests/creators/tests_creators_abstract.py#L37), but it breaks with error message
> TypeError: Can't instantiate abstract class TestCreatorsAbstract with abstract methods setUp

My very wild guess: There seems to be some deep-shit going on because `setUp` is the initiator function.

Can anybody help me or point me to documentation that helps me? Thanks.

Reply:
You would declare a new method `get_selector()` which throws a NotImplemented exception, so all subclasses need to implement it. Then where you need the selector, you call the method.

Reply:
ok, it is implemented as suggested, with two abstract functions:

* `_get_selector()`
* `_get_required_values()`

And optionally can be overridden:

* `_get_standard_values()`
* `_override_configuration()`

Is this how you meant it, @grote?